model/tensorflow/dffml_model_tensorflow/dnnr.py

```diff
@@ -0,0 +1,323 @@
+        Evaluates the accuracy of our model after training using the input repos
+        as test data.
+        """
+        if not os.path.isdir(self.model_dir_path):
+            raise NotADirectoryError("Model not trained")
+        input_fn = await self.evaluate_input_fn(
+            sources, batch_size=20, shuffle=False, epochs=1
+        )
+        metrics = self.model.evaluate(input_fn=input_fn)
+        return Accuracy(1-metrics["loss"]) # 1 - mse
+
+    async def predict(self, repos: AsyncIterator[Repo]) -> AsyncIterator[Repo]:
+        """
+        Uses trained data to make a prediction about the quality of a repo.
+        """
+        
+        if not os.path.isdir(self.model_dir_path):
+            raise NotADirectoryError("Model not trained")
+        # Create the input function
+        input_fn, predict_repo = await self.predict_input_fn(repos)
+        # Makes predictions on 
+        predictions = self.model.predict(input_fn=input_fn)
+   
+        for repo, pred_dict in zip(predict_repo, predictions):
+            repo.predicted(float(pred_dict["predictions"]),float('nan')) # 0,arbitary number to match fun signature -> (value,confidence)
+           
+            yield repo
+
+
+@entry_point("tfdnnr")
+class DNNRegressionModel(Model):
+    """
+    Implemented using Tensorflow's DNNEstimator. Models are saved under the
+    ``directory`` in subdirectories named after the hash of their feature names and hidden layer config.
+
+    usage :
+        * label_name : column_name of target
+        * NOTE : if multiple models are trained then `model-hidden` should be passed to hash to the
+                    model directory
+
+    dffml train \
+          -model tfdnnr \
+          -model-epochs 300 \
+          -model-steps 2000 \
+          -model-label_name {$TARGET_COLUMN_NAME} \
+          -model-hidden 12 40 50 \
+          -sources s=csv \
+          -source-readonly \
+          -source-filename {$TRAIN_DATA_FILENAME}.csv \
+          -features \
+            def:{$feature1}:float:1 \
+            def:{$feature2}:float:1 \
+            def:{$feature3}:float:1 \
+          -log debug
+    
+     dffml accuracy \
+          -model tfdnnr \
+          -model-label_name {$TARGET_COLUMN_NAME} \
+          -model-hidden 12 40 50 \
+          -sources s=csv \
+          -source-readonly \
+          -source-filename {$TRAIN_DATA_FILENAME}.csv \
+          -features \
+            def:{$feature1}:float:1 \
+            def:{$feature2}:float:1 \
+            def:{$feature3}:float:1 \
+          -log debug
+    
+    dffml predict all \
+          -model tfdnnr \
+          -model-label_name {$TARGET_COLUMN_NAME} \
+          -model-hidden 12 40 50 \
+          -sources s=csv \
+          -source-readonly \
+          -source-filename {$TEST_DATA_FILENAME}.csv \
+          -features \
+            def:{$feature1}:float:1 \
+            def:{$feature2}:float:1 \
+            def:{$feature3}:float:1 \
+          -log debug
```

Comment:
We'll need an example file, or take input from stdin for predict: https://github.com/intel/dffml/blame/master/docs/plugins/dffml_model.rst#L363-L371